packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/barrier-selector.jsx

```diff
@@ -0,0 +1,115 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Localize, localize } from '@deriv/translations';
+import { connect } from 'Stores/connect';
+import { CSSTransition } from 'react-transition-group';
+import { Icon, Text, ThemedScrollbars } from '@deriv/components';
+import Fieldset from 'App/Components/Form/fieldset.jsx';
+import BarriersList from './barriers-list.jsx';
+
+const BarrierSelector = ({ barrier_1, onChange, setHoveredBarrier, turbos_barrier_choices }) => {
+    const [selected_barrier, setSelectedBarrier] = React.useState(barrier_1);
+    const [is_barriers_table_expanded, setIsBarriersTableExpanded] = React.useState(false);
+
+    const header_tooltip_text = (
+        <React.Fragment>
+            <div className='trade-container__barriers-tooltip'>
+                <Localize
+                    i18n_default_text="<0>For Long: </0>You will earn profit if the market stays above the entry spot and doesn't cross the barrier."
+                    components={[<Text key={0} weight='bold' size='xxs' />]}
+                />
+            </div>
+            <Localize
+                i18n_default_text="<0>For Short: </0>You will earn profit if the market stays below the entry spot and doesn't cross the barrier."
+                components={[<Text key={0} weight='bold' size='xxs' />]}
+            />
+        </React.Fragment>
+    );
+
+    const toggleBarriersTable = () => setIsBarriersTableExpanded(!is_barriers_table_expanded);
+
+    const changeBarrier = id => {
+        setHoveredBarrier(null);
+        setSelectedBarrier(id);
+        onChange({
+            target: {
+                name: 'barrier_1',
+                value: id,
+            },
+        });
+    };
+
+    const onMouseEnter = id => setHoveredBarrier(id);
+
+    const onMouseLeave = () => setHoveredBarrier(null);
```

Comment:
```suggestion

```
@kate-deriv with my suggestions, we won't need onMouseEnter and onMouseLeave anymore🙂